tests/testing_utils/files_and_dirs.py

```diff
@@ -40,10 +42,11 @@ def temp_local_dir(files: Dict[str, Union[str, bytes]]) -> Generator[Path, None,
         for relpath, contents in files.items():
             path = Path(tmpdir, relpath)
             path.parent.mkdir(parents=True, exist_ok=True)
-            mode = "wb" if isinstance(contents, bytes) else "w"
-            encoding = None if isinstance(contents, bytes) else "UTF-8"
-            with open(path, mode=mode, encoding=encoding) as fh:
-                fh.write(contents)
+            if path.suffix != "":
```

Comment:
This looks like it's attempting to avoid making files that have `None` as their values (as they represent directories), maybe we just explicitly look for that? Directories can have dots in them.

If I understood the use case correctly, my suggestion would be something like:

```python
is_directory = (contents is None)
if not is_directory:
```

Reply:
I like that suggestion, maybe we can even simplify it further. 
```
path = Path(tmpdir, relpath)
...
if path.is_file():
    ...write to file...
```

But, there can be a case where we pass in `contents = None` even if `path` is a file. In that case, I assume we wanted to create a file, but not write anything to it. In that case, I can add a check:
```
if path.is_file:
    if contents is None:
          f = open(path, "x") # creates new file
    else:
          ... write to file as before ...
```

WDYT?